conan_provider.cmake

```diff
@@ -89,11 +89,75 @@ function(detect_cxx_standard CXX_STANDARD)
     endif()
 endfunction()
 
+macro(detect_gnu_libstdcxx)
```

Comment:
I am a bit afraid that this detection could be a bit slow. Are we really sure that this is at least cached, so it doesn't execute at every build, but just when CMake is really modified/reconfigured?

Reply:
According to the cmake documentation:

> The check is only performed once, with the result cached in the variable named by <resultVar>. Every subsequent CMake run will re-use this cached value rather than performing the check again, even if the <code> changes. In order to force the check to be re-evaluated, the variable named by <resultVar> must be manually removed from the cache.


This should be fairly fast as well, but happy to measure the time if needed.